feat: add diagnostic/tracing channel support

tlhunter opened the pull request.

tlhunter committed:
feat: emit diagnostics_channel upon handling request

Eomm committed:
docs: add open-collective (#5216)

dependabot[bot] committed:
chore: Bump actions/upload-artifact from 3 to 4 (#5249)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from 3 to 4.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore: Bump actions/labeler from 4 to 5 (#5248)

Bumps [actions/labeler](https://github.com/actions/labeler) from 4 to 5.
- [Release notes](https://github.com/actions/labeler/releases)
- [Commits](https://github.com/actions/labeler/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/labeler
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Bugs5382 committed:
docs(ecosystem): update fastify-rabbitmq // add fastify-hl7 to ecosystem.md (#5245)

Qard commented on lib/handleRequest.js:
```diff
@@ -133,21 +136,31 @@ function preHandlerCallback (err, request, reply) {
   }
 
   if (result !== undefined) {
+    store.result = result
```
To properly follow the tracing channel spec, the `store.result` value should be set only for the sync branch below and the async branch through `wrapThenable` should set `store.result` to the `payload` object rather than storing that in `store.payload`.

tlhunter replied:
Sounds good. It did feel like I was doing something wrong by swapping the meaning of `.result` depending on if it were an async or sync handler.

climba03003 committed:
chore: update actions/labeler@5 (#5254)

climba03003 committed:
refactor: restrict findRoute exposed property (#5253)

Uzlopak committed:
fix(test): flaky on-listen hook test (#5256)

* fix(test): flaky on-listen hook test

* Apply suggestions from code review

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

* simplify

* use t.equal instead t.ok

---------

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

mcollina committed:
fix: remove unused promise warning in setNotFoundHandler with preHandler (#5258)

mcollina committed:
fix: Always call resource.emitDestroy() (#5228)

* Always call resource.emitDestroy()

Signed-off-by: Matteo Collina <hello@matteocollina.com>

* fixup

Signed-off-by: Matteo Collina <hello@matteocollina.com>

* fixup

Signed-off-by: Matteo Collina <hello@matteocollina.com>

* fixup

Signed-off-by: Matteo Collina <hello@matteocollina.com>

---------

Signed-off-by: Matteo Collina <hello@matteocollina.com>

matthyk committed:
Add missing punctuation in Ecosystem (#5261)

Signed-off-by: Matthias Keckl <53833818+matthyk@users.noreply.github.com>

rlawisch committed:
Remove word repetition (#5260)

This removes a repetition on `route` word (first as plain text, second time as a link to the routes documentation).

Signed-off-by: Rafael Lawisch <lawisch.nh@gmail.com>

Commit:
chore(types): Remove unused imports from fastify.d.ts (#5264)

Signed-off-by: codershiba <155646804+codershiba@users.noreply.github.com>

Commit:
chore(license): Update licensing year (#5266)

Commit:
chore(docs): Add clarification about `fastify.setErrorHandler()` (#5265)

Commit:
refactor: deprecate `Reply#getResponseTime()` in favour of `Reply#elapsedTime` (#5263)

Fdawgs committed:
chore: remove `www.` from `fastify.dev` urls (#5270)

dancastillo committed:
feat: expose method for setGenReqId on FastifyInstance (#5259)

dancastillo committed:
feat: align fastify.hasRoute to fmw.hasRoute (#5102)

bienzaaron committed:
fix: ensure `onListen` hooks are called when they should be (#5273)

* fix: ensure `onListen` hooks is called for nested encapsulation contexts

* also ensure `onListen` runs for all peer contexts

Commit:
docs: re-word clarification about `setErrorHandler()` (#5269)

Signed-off-by: codershiba <155646804+codershiba@users.noreply.github.com>

Fdawgs committed:
docs(ecosystem): remove unspported package (#5278)

BoscoDomingo committed:
docs: Fix Pino docs link (#5284)

* chore: Fix Pino docs link and refactor to Semantic Line Breaks

More info here: https://sembr.org/

I updated the Pino docs link as the previous one was broken and took advantage to use Semantic Line Breaks for easier updates in the future without changing the formatting

Signed-off-by: Bosco Domingo <46006784+BoscoDomingo@users.noreply.github.com>

* docs: Undo unnecessary changes from previous commit

This reverts commit 099614ecd7c2368ecfd26d24d02f5dae5914cae5.

---------

Signed-off-by: Bosco Domingo <46006784+BoscoDomingo@users.noreply.github.com>

Eomm committed:
chore: add github sponsor (#5293)

* chore: update funding

Signed-off-by: Manuel Spigolon <behemoth89@gmail.com>

* chore: update EXPENSE_POLICY

Signed-off-by: Manuel Spigolon <behemoth89@gmail.com>

* chore: update SPONSORS.md

Signed-off-by: Manuel Spigolon <behemoth89@gmail.com>

* Update SPONSORS.md

Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>
Signed-off-by: Manuel Spigolon <behemoth89@gmail.com>

---------

Signed-off-by: Manuel Spigolon <behemoth89@gmail.com>
Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>

yoav0gal committed:
docs(ecosystem): adds fastify-sqlite-typed to the community plugins list (#5288)

atilagulers committed:
docs: add ESM usage example in Getting Started (#5294)

Fdawgs committed:
docs: repoint readers to shared `.github` files (#5268)

climba03003 committed:
feat: Web Stream API (#5286)

* feat: Web Stream API

* chore: update error PR number

* test: skip test when not available

* test: exit when skip test

* test: use Readable.toWeb instead

* docs: update ToC and onSend hook

* refactor: toString

* refactor: direct for-of payload.headers

* test: ensure compatibility of third-party Response

* refactor: reduce toString call

Eomm committed:
chore: sync generated code (#5295)

Eomm committed:
Bumped v4.26.0

yoav0gal committed:
docs(ecosystem): adds fastify-hana to the community plugins list (#5289)

* docs(ecosystem): adds fastify-hana to the community plugins list

* docs(Ecosystem): changed fastify-hana placemet to fit in a-z order

* docs(ecosystem): adds fastify-hana to the community plugins list

Signed-off-by: Yoav Gal <45267791+yoav0gal@users.noreply.github.com>

---------

Signed-off-by: Yoav Gal <45267791+yoav0gal@users.noreply.github.com>

mscottnelson committed:
fix misattributed property parent in deprecation warning: request.elapsedTime -> reply.elapsedTime (#5299)

dependabot[bot] committed:
chore: Bump lycheeverse/lychee-action from 1.8.0 to 1.9.3 (#5300)

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 1.8.0 to 1.9.3.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/ec3ed119d4f44ad2673a7232460dc7dff59d2421...c053181aa0c3d17606addfe97a9075a32723548a)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore: Bump actions/dependency-review-action from 3 to 4 (#5301)

Bumps [actions/dependency-review-action](https://github.com/actions/dependency-review-action) from 3 to 4.
- [Release notes](https://github.com/actions/dependency-review-action/releases)
- [Commits](https://github.com/actions/dependency-review-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/dependency-review-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Fdawgs committed:
chore(.gitignore): add .tap/ dir (#5303)

Signed-off-by: Frazer Smith <frazer.dev@outlook.com>

mcollina committed:
fix: amend error codes for latest avvio v8.3.0 (#5309)

* Fix tests for latest avvio

Signed-off-by: Matteo Collina <hello@matteocollina.com>

* fixup

Signed-off-by: Matteo Collina <hello@matteocollina.com>

---------

Signed-off-by: Matteo Collina <hello@matteocollina.com>

rozzilla committed:
fix(types): Request route options url add undefined (#5307)

* fix(types): Request route options url add undefined

* Add test

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

---------

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>
Co-authored-by: Matteo Collina <hello@matteocollina.com>

jsumners committed:
Add docs for tracing warnings (#5310)

mcollina committed:
Bumped v4.26.1

Signed-off-by: Matteo Collina <hello@matteocollina.com>

lirantal committed:
fix: typo in module exports

Signed-off-by: Liran Tal <liran.tal@gmail.com>

rozzilla committed:
docs(ts): Fix links (#5308)

* chore(ts): Fix links

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

* Use absolute URLs

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

* Revert

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

---------

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

Uzlopak committed:
fix: cb is not a function at fallbackErrorHandler (#5317)

* fix: cb is not a function at fallbackErrorHandler

* Update docs/Reference/Errors.md

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

---------

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

lirantal committed:
feat: add a Firebase Functions step by step guide (#5318)

* feat: add a Firebase Functions step by step guide

This PR adds a new section to the Serverless Guide on the website that explains how to use Fastify as the wrapper around Firebase's own onRequest HTTP handler

Signed-off-by: Liran Tal <liran.tal@gmail.com>

* Update docs/Guides/Serverless.md

Co-authored-by: James Sumners <321201+jsumners@users.noreply.github.com>
Signed-off-by: Liran Tal <liran.tal@gmail.com>

* Update docs/Guides/Serverless.md

Co-authored-by: James Sumners <321201+jsumners@users.noreply.github.com>
Signed-off-by: Liran Tal <liran.tal@gmail.com>

* Update docs/Guides/Serverless.md

Co-authored-by: James Sumners <321201+jsumners@users.noreply.github.com>
Signed-off-by: Liran Tal <liran.tal@gmail.com>

* Update docs/Guides/Serverless.md

Co-authored-by: James Sumners <321201+jsumners@users.noreply.github.com>
Signed-off-by: Liran Tal <liran.tal@gmail.com>

---------

Signed-off-by: Liran Tal <liran.tal@gmail.com>
Co-authored-by: James Sumners <321201+jsumners@users.noreply.github.com>

climba03003 committed:
Merge branch 'main' into next

# Conflicts:
#	docs/Reference/Errors.md
#	fastify.js
#	lib/warnings.js
#	package.json
#	test/genReqId.test.js
#	test/internals/errors.test.js
#	types/instance.d.ts

dancastillo committed:
feat: set useSemicolonDefault default option to false (#5320)

* feat: set useSemicolonDefault default option to false

* feat: add migration guide for v5

* Update docs/Guides/Migration-Guide-V5.md

Co-authored-by: Frazer Smith <frazer.dev@outlook.com>
Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

---------

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>
Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>
Co-authored-by: Frazer Smith <frazer.dev@outlook.com>

gurgunday committed:
fix type (#5330)

gurgunday committed:
perf: use FifoMap (#5331)

matthyk committed:
Update Ecosystem.md (#5336)

climba03003 committed:
Merge branch 'main' into next

gurgunday committed:
feat: make contentTypeParser's existingParser check more strict (#5329)

climba03003 committed:
feat: loosen content-type checking (#4450)

matthyk committed:
types: Export preClose hook types (#5335)

nuhman committed:
Establish database connection in migration code (#5339)

dependabot[bot] committed:
chore: Bump pnpm/action-setup from 2 to 3 (#5341)

Bumps [pnpm/action-setup](https://github.com/pnpm/action-setup) from 2 to 3.
- [Release notes](https://github.com/pnpm/action-setup/releases)
- [Commits](https://github.com/pnpm/action-setup/compare/v2...v3)

---
updated-dependencies:
- dependency-name: pnpm/action-setup
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore: Bump xt0rted/markdownlint-problem-matcher from 2.0.0 to 3.0.0 (#5342)

Bumps [xt0rted/markdownlint-problem-matcher](https://github.com/xt0rted/markdownlint-problem-matcher) from 2.0.0 to 3.0.0.
- [Release notes](https://github.com/xt0rted/markdownlint-problem-matcher/releases)
- [Changelog](https://github.com/xt0rted/markdownlint-problem-matcher/blob/main/CHANGELOG.md)
- [Commits](https://github.com/xt0rted/markdownlint-problem-matcher/compare/98d94724052d20ca2e06c091f202e4c66c3c59fb...1a5fabfb577370cfdf5af944d418e4be3ea06f27)

---
updated-dependencies:
- dependency-name: xt0rted/markdownlint-problem-matcher
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

mcollina committed:
Bumped v4.26.2

Signed-off-by: Matteo Collina <hello@matteocollina.com>

Tony133 committed:
docs(request): update request page (#5343)

riosje committed:
ci: Add Nsolid runtime to CI and Integration tests (#5332)

* ci: Add Nsolid CI and Integration tests as separate pipelines

Signed-off-by: Jefferson <jefferson.rios.caro@gmail.com>

Shyam-Chen committed:
docs(Ecosystem): add fastify-i18n, vite-plugin-fastify, and vite-plugin-fastify-routes (#5346)

Co-authored-by: Carlos Fuentes <me@metcoder.dev>

jorgevrgs committed:
docs(ecosystem): update x-ray community plugin link (#5350)

* docs: update x-ray community plugin link

* docs: update package name and description

* docs: sort in alphabetical order

10xLaCroixDrinker committed:
docs: spelling corrections (#5349)

* docs: spelling corrections

* docs: spelling corrections

* test: correct spelling in specs and assertions

Co-authored-by: Frazer Smith <frazer.dev@outlook.com>
Signed-off-by: Jamie King <hello@jamieking.me>

---------

Signed-off-by: Jamie King <hello@jamieking.me>
Co-authored-by: Frazer Smith <frazer.dev@outlook.com>

BrianValente committed:
Fix `config` type in RouteShorthandOptions

melroy89 committed:
Remove ES module warning (#5361)

https://github.com/fastify/fastify/issues/4241 seems to closed.

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

flower-of-the-bridges committed:
docs(ecosystem): add fastify-msgraph-change-notifications-webhook to community plugins (#5360)

* Update Ecosystem.md

add fastify-msgraph-change-notifications-webhook to community plugins

Signed-off-by: Giovanni Fiordeponti <38134891+flower-of-the-bridges@users.noreply.github.com>

* fix: markdown lint in Ecosystem.md

Signed-off-by: Giovanni Fiordeponti <38134891+flower-of-the-bridges@users.noreply.github.com>

---------

Signed-off-by: Giovanni Fiordeponti <38134891+flower-of-the-bridges@users.noreply.github.com>

melroy89 committed:
docs: Add GitHub links to Type Providers + mention Zod (#5365)

* Add GitHub links and mention Zod

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Fix text for TypeBox

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Create links around the package names

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Also list the wrapper packages on Type Provider page + consistency 

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Update docs/Reference/TypeScript.md

Co-authored-by: Frazer Smith <frazer.dev@outlook.com>
Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Add 3rd party 

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Moved 3rs party line

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Fix linter?

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* Make linter happy, plz

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

---------

Signed-off-by: Melroy van den Berg <melroy@melroy.org>
Co-authored-by: Frazer Smith <frazer.dev@outlook.com>

gurgunday committed:
feat: hasParser, only accept string or RegExp (#5372)

aivopaas committed:
docs: fix table layout (#5374)

Signed-off-by: Aivo Paas <aivopaas@gmail.com>

dependabot[bot] committed:
chore: Bump pnpm/action-setup from 2 to 3 (#5381)

Bumps [pnpm/action-setup](https://github.com/pnpm/action-setup) from 2 to 3.
- [Release notes](https://github.com/pnpm/action-setup/releases)
- [Commits](https://github.com/pnpm/action-setup/compare/v2.0.0...v3)

---
updated-dependencies:
- dependency-name: pnpm/action-setup
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Tony133 committed:
docs(typescript): update generic constraints link (#5379)

hpx7 committed:
types: narrow code reply type based on schema (#5380)

* narrow code reply type based on schema

* more assertions

melroy89 committed:
fix: errorCodes import using ESM (#5390)

* Fix errorCodes for ESM

Moving fastify errorCodes below module.exports = fastifty and also export it

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

* test: Add errorCodes test for ESM

---------

Signed-off-by: Melroy van den Berg <melroy@melroy.org>

melroy89 committed:
fix: Extend tap with .test.mjs tests & Rename existing ESM test (#5391)

* Fix other ESM test case. And add comments

* Add extend tap with test.mjs files

Tony133 committed:
docs(typescript): update example https server (#5389)

gurgunday committed:
perf: limit search space for contentType (#5400)

Tony133 committed:
docs(typescript): update snippet code (#5406)

gurgunday committed:
fix bench co (#5414)

gurgunday committed:
fix: `remove-label` job uses the wrong repo (#5415)

* fix: remove label gets passed the wrong repo information

* add cithm too

* use github event

* add permissions

* use list form

gurgunday committed:
only run citgm if citgm-core-plugins label is set (#5416)

johaven committed:
feat: adds webdav methods that require body & content type parsing (#5411)

* feat: adds webdav methods that require body & content type parsing

* fix: lint error

* fix: UNLOCK method does not require body

* fix: lock,propfind,proppatch tests with content type

* fix: Routes.md

* fix: tests coverage

tmcw committed:
docs: improve onError docs by specifying what the error handler is (#5358)

* docs: Improve onError docs by specifying what the error handler is

* Apply suggestions from code review

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>

---------

Signed-off-by: Aras Abbasi <aras.abbasi@googlemail.com>
Co-authored-by: Aras Abbasi <aras.abbasi@googlemail.com>

Eomm committed:
chore: add new sponsor (#5424)

climba03003 committed:
Merge branch 'main' into next

# Conflicts:
#	docs/Reference/Server.md
#	fastify.js
#	package.json

rozzilla committed:
chore(ecosystem): Add Fastify asyncforge plugin (#5429)

* chore(ecosystem): Add Fastify asyncforge plugin

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

* Update Ecosystem.md

---------

Signed-off-by: Roberto Bianchi <roberto.bianchi@spendesk.com>

remidewitte committed:
`reply.getSerializationFunction` can return `undefined` (#5384)

dependabot[bot] committed:
chore: Bump pino from 8.21.0 to 9.0.0 in the dependencies-major group (#5431)

Bumps the dependencies-major group with 1 update: [pino](https://github.com/pinojs/pino).


Updates `pino` from 8.21.0 to 9.0.0
- [Release notes](https://github.com/pinojs/pino/releases)
- [Commits](https://github.com/pinojs/pino/compare/v8.21.0...v9.0.0)

---
updated-dependencies:
- dependency-name: pino
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: dependencies-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

gurgunday committed:
exclude node 14 and 16 on macos (#5433)

gurgunday committed:
perf: update method matching (#5419)

* update method matching implementation

* push back string comparison

* extract encoding

* add missing method test

* check for transferEncoding as well

* skip instead of removing test

* Update test/delete.test.js

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>
Signed-off-by: Gürgün Dayıoğlu <hey@gurgun.day>

* Update test/method-missing.test.js

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>
Signed-off-by: Gürgün Dayıoğlu <hey@gurgun.day>

---------

Signed-off-by: Gürgün Dayıoğlu <hey@gurgun.day>
Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>

dependabot[bot] committed:
chore: Bump lycheeverse/lychee-action from 1.9.3 to 1.10.0 (#5436)

Bumps [lycheeverse/lychee-action](https://github.com/lycheeverse/lychee-action) from 1.9.3 to 1.10.0.
- [Release notes](https://github.com/lycheeverse/lychee-action/releases)
- [Commits](https://github.com/lycheeverse/lychee-action/compare/c053181aa0c3d17606addfe97a9075a32723548a...2b973e86fc7b1f6b36a93795fe2c9c6ae1118621)

---
updated-dependencies:
- dependency-name: lycheeverse/lychee-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Tony133 committed:
docs: update indentation on snippet code (#5418)

Fdawgs committed:
docs(guides/abort):  suggest explicit use of the `aborted` property (#5438)

SamSalvatico committed:
feat: disable request logging (#5435)

* fix(error-handler): check for kDisableRequestLogging before logging

* fix(error-handler): updated tests

* fix(fourOhfour): check for logging disabled before log 404

* fix(fourOhfour): managed disableRequestLogging

* fixed test for server listening message

* Updated docs related to disableRequestLogging

* fixed typo in docs

gurgunday committed:
update readme (#5442)

keyservlad committed:
feat: add mkcalendar and report methods (#5439)

Co-authored-by: aguilhamat <arnaud.guilhamat.ext@orange.com>
Co-authored-by: Carlos Fuentes <me@metcoder.dev>

Commit:
chore: updated dependencies to latest versions (#5422)

* updated all the packages which required minor / patch version bumps

* updated:

@typescript-eslint/eslint-plugin
@typescript-eslint/parser

* updated:

c8
ajv-formats

* updated:

markdownlint-cli2

* fully working eslint config

* removed some unused deps

* removed some unused deps

* updated:

tsd
undici

---------

Co-authored-by: Giovanni Bucci <giovanni.bucci@nearform.com>

jsumners committed:
Merge branch 'main' into next

jsumners committed:
chore: revert eslint changes

jsumners committed:
chore: restore #5419

jsumners committed:
chore: update node version in workflows

jsumners committed:
chore: alt runtimes do not support 22 yet

jsumners committed:
chore: update benchmark ci

mcollina committed:
feat: handle synchronous errors in errorHandler (#5445)

james-kaguru committed:
updated the docs to match v3 requirements

nflaig committed:
types: request route schema might be undefined (#5394)

* types: request route schema might be undefined

* Update types/request.d.ts

Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>
Signed-off-by: Nico Flaig <nflaig@protonmail.com>

* Comment why it might be undefined / empty

Signed-off-by: Nico Flaig <nflaig@protonmail.com>

---------

Signed-off-by: Nico Flaig <nflaig@protonmail.com>
Co-authored-by: Manuel Spigolon <behemoth89@gmail.com>

mcollina committed:
Bumped v4.27.0

Signed-off-by: Matteo Collina <hello@matteocollina.com>

mcollina committed:
Merge branch 'main' into next

Signed-off-by: Matteo Collina <hello@matteocollina.com>

mcollina committed:
Updated README

Signed-off-by: Matteo Collina <hello@matteocollina.com>

mcollina committed:
Bumped v5.0.0-alpha.1

Signed-off-by: Matteo Collina <hello@matteocollina.com>

tlhunter committed:
use TracingChannel#hasSubscribers

gurgunday committed:
chore: remove dep (#5454)

gurgunday committed:
add shorthand types (#5455)

dmkng committed:
Optimize resolving X-Forwarded-For addresses

Signed-off-by: Susan <25956407+dmkng@users.noreply.github.com>

dmkng committed:
Replaced proxyAddr with proxyAddr.all

Signed-off-by: Susan <25956407+dmkng@users.noreply.github.com>

dmkng committed:
Made argument absence look cleaner

Co-authored-by: Gürgün Dayıoğlu <hey@gurgun.day>
Signed-off-by: Susan <25956407+dmkng@users.noreply.github.com>

Eomm committed:
fix: ci labeler (#5463)

monish001 committed:
docs(guides/database): correct typo (#5461)

* Fixed a typo in documentation in Database.md

From: `Each file an migrations folder need`
To: `Each file in migrations folder need`

Signed-off-by: Monish <monish.gupta1@gmail.com>

* Update docs/Guides/Database.md

Co-authored-by: Frazer Smith <frazer.dev@icloud.com>
Signed-off-by: Monish <monish.gupta1@gmail.com>

---------

Signed-off-by: Monish <monish.gupta1@gmail.com>
Co-authored-by: Frazer Smith <frazer.dev@icloud.com>

domdomegg committed:
test: add tests for error handling (#5451)

* Add tests for error handling

* Fix t.plan counts

* Fix t.plan counts

dmkng committed:
Changed the comment to be more accurate

Signed-off-by: Susan <25956407+dmkng@users.noreply.github.com>

BrianValente committed:
Add test for `config` type in RouteShorthandOptions

Cadienvan committed:
docs(reference/routes): fix example on constraints (#5468)

Signed-off-by: Michael Di Prisco <Cadienvan@users.noreply.github.com>

tlhunter committed:
update dc-polyfill to fix a hang in c8 lib

Eomm committed:
feat: request and reply decorators can not be a reference type (#5462)

Tony133 committed:
docs: update indentation on type providers section (#5474)

mcollina committed:
feat: log all available addresses if listening host is 0.0.0.0 (#5476)

gurgunday committed:
chore: remove node 18 from CI (#5481)

gurgunday committed:
feat: writeEarlyHints (#5480)

Tony133 committed:
docs: update indentation and code snippet in the routes section (#5482)

gurgunday committed:
refactor: change `reply.redirect()` signature (#5483)

* feat: change `reply.redirect()` signature

* feat: change `reply.redirect()` signature

* docs

* docs

* update message

* fix deprecation

* update message

Tony133 committed:
docs: update indentation and code snippet in the type providers section (#5485)
